src/main/scala/ai/starlake/schema/handlers/HdfsStorageHandler.scala

```diff
@@ -243,40 +243,55 @@ class HdfsStorageHandler(fileSystem: String)(implicit
   }
 
   /** Read a UTF-8 text file into a string used to load yml configuration files
-    *
-    * @param path
-    *   : Absolute file path
-    * @return
-    *   file content as a string
-    */
+   *
+   * @param path
+   * : Absolute file path
+   * @return
+   * file content as a string
+   */
   def read(path: Path, charset: Charset = StandardCharsets.UTF_8): String = {
```

Comment:
We load the whole file in memory right ?

Reply:
Just refactored the code and added readAndExecute.

Off course I tried not to read the whole file before getting the headers so I think there is no such limit.